play-json/shared/src/main/scala/JsResult.scala

```diff
@@ -197,8 +197,33 @@ sealed trait JsResult[+A] { self =>
 }
 
 object JsResult {
+  import scala.util.{ Failure, Try, Success }
   import play.api.libs.functional._
 
+  case class Exception(cause: JsError)
+    extends IllegalArgumentException(Json stringify JsError.toJson(cause))
```

Comment:
"typical Scala user would expect that either." That's not the point, as the Scala dev will sees it's a `JsError.Exception`, which is quite explicit for me.

The `.toJson` is not useful, as the `Exception` will be in case of convertion from JSON validation to `Try`. The `.toJson` is useful when considering the error as validation error.